sentry-core/src/performance.rs

```diff
@@ -273,6 +300,14 @@ pub(crate) struct TransactionInner {
     pub(crate) transaction: Option<protocol::Transaction<'static>>,
 }
 
+#[derive(Default)]
+#[cfg(all(feature = "profiling", not(target_os = "windows")))]
+struct ProfileInner {
+    transaction_id: String,
+    #[cfg(all(feature = "profiling", not(target_os = "windows")))]
+    profiler_guard: Option<pprof::ProfilerGuard<'static>>,
+}
```

Comment:
The whole struct is already `cfg`-guarded, so no need to have the member as well.

Also, I would rather make the whole struct an `Option`, instead of an `Option` member and an empty string for the transaction.

In general, I would remove the mutable global altogether, and rather attach the `Option<ProfilesGuard>` onto the transaction. That way, you wouldn’t have to take the lock and do a transaction_id comparison on *every* transaction finish.

I would still have one global `AtomicBool` that says `PROFILER_RUNNING`.

You should be able to use a [compare_exchange](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html#method.compare_exchange)(false, true), then try to start your profiler. If that fails for some reason, just `store(false)` back into it.
Similarly, to stop profiling, you will only work on the `Transaction` itself, and if you *do* have an associated `ProfilerGuard`, you can stop it and `store(false)` as well. That way, you would only have a single CAS instead of potentially longer running locks.

Reply:
> The whole struct is already cfg-guarded, so no need to have the member as well

yep, I definitely forgot to remove it when I moved it at the struct level.

Reply:
> In general, I would remove the mutable global altogether, and rather attach the Option<ProfilesGuard> onto the transaction. That way, you wouldn’t have to take the lock and do a transaction_id comparison on every transaction finish

I would love such implementation!! 
unfortunately, the `build` [method](https://docs.rs/pprof/latest/pprof/struct.ProfilerGuardBuilder.html#method.build) of `ProfilerGuardBuilder` returns a `ProfilerGuard<'static>`, which does not implement the `clone` trait and so I cannot add it as an Option field in the `Transaction` struct defined in `sentry-core/performance.rs` since `Transaction` derives `Clone`.